examples/TransactionGenerator.hs

```diff
@@ -299,92 +555,194 @@ mainInfo =
+      mgr <- newManager defaultManagerSettings
+      withBaseHandleBackend "transaction-generator" mgr (defconfig ^. U.logConfigBackend) $ \baseBackend -> do
+        let loggerBackend = logHandles [] baseBackend
+        withLogger (U._logConfigLogger defconfig) loggerBackend $
+          runLoggerT $
+            case _scriptCommand config of
+             DeployContracts contracts mtime -> liftIO $ if null contracts
+                  then loadContracts mtime (initAdminKeysetContract : defaultContractLoaders) config
+                  else loadContracts mtime (initAdminKeysetContract : fmap createLoader contracts) config
+             RunStandardContracts distribution mtime -> sendTransactions mtime config distribution
+             RunSimpleExpressions distribution mtime -> sendSimpleExpressions mtime config distribution
+             PollRequestKeys requestKeys mtime -> liftIO $ pollRequestKeys mtime (RequestKeys (map (RequestKey . Hash) requestKeys)) config
+             ListenerRequestKey requestKey mtime -> liftIO $ listenerRequestKey mtime (ListenerRequest (RequestKey $ Hash requestKey)) config
+
+-- TODO: This is here for when a user wishes to deploy their own
 -- contract to chainweb. We will have to carefully consider which
 -- chain we'd like to send the contract to.
 
 -- TODO: This function should also incorporate a user's keyset as well
 -- if it is given.
-createLoader :: FilePath -> ContractLoader
-createLoader contractName = \_kp -> do
-    theCode <- readFile contractName
-    adminKeyset <- testSomeKeyPairs
-    let theData = object ["admin-keyset" .= fmap formatB16PubKey adminKeyset]
-    mkExec theCode theData def adminKeyset Nothing
+createLoader :: ContractName -> ContractLoader
+createLoader (ContractName contractName) meta kp = do
+  theCode <- readFile (contractName <> ".pact")
+  adminKeyset <- testSomeKeyPairs
+  -- TODO: theData may change later
+  let theData = object
+                ["admin-keyset" .= fmap formatB16PubKey adminKeyset
+                , T.append (T.pack contractName) "-keyset" .= fmap formatB16PubKey kp]
+  mkExec theCode theData meta adminKeyset Nothing
 
 defaultContractLoaders :: [ContractLoader]
-defaultContractLoaders = take numContracts $ [helloWorldContractLoader, simplePaymentsContractLoader, cryptoCritterContractLoader]
+defaultContractLoaders = take numContracts
+  [helloWorldContractLoader
+  , simplePaymentsContractLoader
+  , cryptoCritterContractLoader]
+  -- Remember coin contract is already loaded.
+
 -- add this back in later
 -- , commercialPaperContractLoader
 
 -- We are just going to work with some contracts at this point in time.
 numContracts :: Int
 numContracts = 2
 
-send :: SubmitBatch -> ClientM RequestKeys
-poll :: Poll -> ClientM PollResponses
-listen :: ListenerRequest -> ClientM ApiResult
-local :: Command Text -> ClientM (CommandSuccess Value)
-
-send :<|> poll :<|> listen :<|> local = client (Proxy :: Proxy (PactApi_))
+type TheApi
```

Comment:
Why do we need to redeclare the API here?  We should be able to re-use the upstream definition and avoid the repetition.

Reply:
While I could use `somePactApi` which has type `somePactApi :: ChainwebVersion -> ChainId -> SomeApi`, that would require unpacking the data type SomeApi and it does not seem like it is possible to retrieve the appropriate Proxy value when unpacking it. Hopefully, someone can chime in and point out if I'm wrong.

Reply:
...and `Pact.Server.API`

Reply:
I've addressed this by making `send`, `poll` and `listen` top level commands (they're out of the `Reader`). They, of course, still must the chainweb version and the chain id as arguments in the call (in addition to their usual arguments). I think this solution is cleaner than mucking around with the `BaseUrl` object.

Reply:
I've actually addressed this in another manner.